django/core/validators.py

```diff
@@ -87,7 +87,7 @@ class URLValidator(RegexValidator):
         r'^(?:[a-z0-9.+-]*)://'  # scheme is validated separately
         r'(?:[^\s:@/]+(?::[^\s:@/]*)?@)?'  # user:pass authentication
         r'(?:' + ipv4_re + '|' + ipv6_re + '|' + host_re + ')'
-        r'(?::\d{2,5})?'  # port
+        r'(?::\d+)?'  # port
```

Comment:
Using `\d` instead of `\d{1,5}` is risky and would create a security issue :fire: . The previous solutions raises a `ValidationError` almost instantly on crafted URLs, e.g.
```pycon
>>> from django.core.validators import URLValidator
>>> x = 'https://code.djangoproject.com:' + '9' * (2**24)
>>> URLValidator()(x)
...
  File "/django/django/core/validators.py", line 51, in __call__
    raise ValidationError(self.message, code=self.code, params={'value': value})
django.core.exceptions.ValidationError: <exception str() failed>
```
with this patch I waited few minutes and gave up.

Reply:
Should I add `'https://code.djangoproject.com:' + '9' * (2**24)` to the test case, so other developers can realize this case?

Reply:
it's a very, very long URL.(~16MB in size).